internal/storage/fs/git/reference_resolvers_test.go

```diff
@@ -0,0 +1,146 @@
+package git
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-git/go-git/v5/plumbing/object"
+
+	"github.com/go-git/go-billy/v5/memfs"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStaticResolver(t *testing.T) {
+	t.Run("should resolve static references correctly", func(t *testing.T) {
+		repo := newGitRepo(t)
+		resolver := StaticResolver()
+
+		commitHash := repo.createCommit(t)
+		resolvedHash, err := resolver(repo.repo, "main")
+
+		require.NoError(t, err)
+		require.Equal(t, commitHash, resolvedHash)
+
+		repo.checkout(t, "new-branch")
+
+		commitHash = repo.createCommit(t)
+		resolvedHash, err = resolver(repo.repo, "new-branch")
+
+		require.NoError(t, err)
+		require.Equal(t, commitHash, resolvedHash)
+	})
+}
+
+func TestSemverResolver(t *testing.T) {
+	t.Run("should resolve semver tags correctly when the reference is a constraint", func(t *testing.T) {
+		repo := newGitRepo(t)
+		resolver := SemverResolver()
+		constraint := "v0.1.*"
+
+		commitHash := repo.createCommit(t)
+		repo.createTag(t, "v0.1.0", commitHash)
+
+		resolvedHash, err := resolver(repo.repo, constraint)
+
+		require.NoError(t, err)
+		require.Equal(t, commitHash, resolvedHash)
+
+		// When committing again and creating a new tag respecting the constraint should resolve nicely
+		commitHash = repo.createCommit(t)
```

Comment:
can we add one more commit above this to test that it doesn't fail on non-semver tags too?

ex: create a tag like `foo` and ensure that it still resolves to the expected correct `v0.1.4`

Reply:
@thepabloaguilar interesting. I think people would quite like to support both strict and 'lax', is there a reason why you want to only enforce strict?

Reply:
Sincerally idk, it's just a thing that came in my mind! 😆

Reply:
I've pushed the changes